lib/resources/yum.rb

```diff
@@ -121,22 +120,25 @@ def shortname(id)
     def info
       return @cache if defined?(@cache)
       selection = @yum.repositories.select { |e| e['id'] == @reponame || shortname(e['id']) == @reponame }
-      @cache = selection[0] if !selection.nil? && selection.length == 1
+      @cache = selection.empty? ? {} : selection.first
       @cache
     end
 
     def exist?
-      !info.nil?
+      !info.empty?
     end
 
     def enabled?
-      repo = info
-      return false if repo.nil?
+      return false unless exist?
       info['status'] == 'enabled'
     end
 
+    def method_missing(name)
```

Comment:
We try to avoid method_missing whenever possible. Do we know all possible properties? Maybe we use something like:

```
    %w{
      baseurl
    }.each do |m|
      define_method m.to_sym do |*args|
        info[m.to_s]
      end
    end
```